File: Client.Vue/wms-client/src/types/refs.ts
export interface ProductRef {
  id: string
  sku: string
  name: string
  unitPrice: number
}

export interface LocationRef {
  id: string
  code: string
  address: string
}

export interface LotRef {
  id: string
  number: string
  expirationDate: string | null
}

File: Api.ASP/Wms.Application/Refs/Refs.cs
using Wms.Domain.Enums;

namespace Wms.Application.Refs;

public sealed record ProductRef(Guid Id, string Sku, string Name, decimal UnitPrice);

public sealed record LocationRef(Guid Id, string Code, string Address);

public sealed record LotRef(Guid Id, string Number, DateOnly? ExpirationDate);

public sealed record HandlingUnitRef(Guid Id, string Code, HandlingUnitType Type);
